ethereum/evmtool/src/main/java/org/hyperledger/besu/evmtool/EvmToolCommand.java

```diff
@@ -291,18 +281,90 @@ public void run() {
                     .put("timens", lastTime)
                     .put("time", lastTime / 1000)
-                    .put("gasTotal", "0x" + Long.toHexString(evmGas)));
+                    .put(
+                        "gasTotal",
+                        "0x" + Long.toHexString(evmGas + intrinsicGasCost) + accessListCost));
           }
         }
         lastTime = stopwatch.elapsed().toNanos();
         stopwatch.reset();
+        if (showJsonAlloc && lastLoop) {
+          updater.commit();
+          out.println("{");
+          component
+              .getWorldState()
+              .streamAccounts(Bytes32.ZERO, Integer.MAX_VALUE)
+              .sorted(Comparator.comparing(o -> o.getAddress().get().toHexString()))
+              .forEach(
+                  account -> {
+                    out.println(
+                        " \""
+                            + account.getAddress().map(Address::toHexString).orElse("-")
+                            + "\": {");
+                    if (account.getCode() != null && account.getCode().size() > 0) {
+                      out.println("  \"code\": \"" + account.getCode().toHexString() + "\",");
+                    }
+                    var storateEntries =
+                        account.storageEntriesFrom(Bytes32.ZERO, Integer.MAX_VALUE);
+                    if (!storateEntries.isEmpty()) {
+                      out.println("  \"storage\": {");
+                      out.println(
+                          STORAGE_JOINER.join(
+                              storateEntries.values().stream()
+                                  .map(
+                                      accountStorageEntry ->
+                                          "   \""
+                                              + accountStorageEntry
+                                                  .getKey()
+                                                  .map(UInt256::toHexString)
+                                                  .orElse("-")
+                                              + "\": \""
+                                              + accountStorageEntry.getValue().toHexString()
+                                              + "\"")
+                                  .collect(Collectors.toList())));
+                      out.println("  },");
+                    }
+                    out.print("  \"balance\": \"" + account.getBalance().toShortHexString() + "\"");
+                    if (account.getNonce() > 0) {
+                      out.println(",");
+                      out.println(
+                          "  \"nonce\": \""
+                              + Bytes.ofUnsignedLong(account.getNonce()).toShortHexString()
+                              + "\"");
+                    } else {
+                      out.println();
+                    }
+                    out.println(" },");
+                  });
+          out.println("}");
+        }
```

Comment:
can we deduplicate this code?

Reply:
One is PrintWriter, one is PrintStream, but by deeply wrapping the printstream it can be unified.